test/builtin_dataset_mocks.py

```diff
@@ -111,21 +113,20 @@ def parametrize_dataset_mocks(*dataset_mocks, marks=None):
         raise pytest.UsageError()
 
     return pytest.mark.parametrize(
-        ("dataset_mock", "config"),
+        ("dataset_mock", "options"),
         [
-            pytest.param(dataset_mock, config, id=config_id(name, config), marks=marks.get(name, ()))
+            pytest.param(dataset_mock, options, id=config_id(name, options), marks=marks.get(name, ()))
             for name, dataset_mock in dataset_mocks.items()
-            for config in dataset_mock.configs
+            for options in dataset_mock.configs
         ],
     )
 
 
-DATASET_MOCKS = {}
+DATASET_MOCKS = []
 
 
 def register_mock(fn):
-    name = fn.__name__.replace("_", "-")
-    DATASET_MOCKS[name] = DatasetMock(name, fn)
+    # TODO: remove this decorator after all datasets have been migrated
```

Comment:
The decorator is a no-op with this change and thus allows us to go one by one for each dataset without touching everything in the first PR.